subsys/net/ip/tcp.c

```diff
@@ -2241,8 +2241,15 @@ static uint32_t tcpv6_init_isn(struct in6_addr *saddr,
 	memcpy(buf.key, unique_key, sizeof(buf.key));
 
 #if defined(CONFIG_NET_TCP_ISN_RFC6528)
+#if defined(CONFIG_BUILD_WITH_TFM)
```

Comment:
Instead of directly checking on TFM, could you create an option to `Kconfig.tcp` that we could use here?

Reply:
There is a reason `CONFIG_BUILD_WITH_TFM` is used here. We have made an assumption that PSA crypto APIs can't be supported unless Mbed TLS supports **Native Internal Trusted Storage**, but this is beyond the scope of this small fixup.

As stated before, it would be possible to not have this if-case at all if we use psa_hash_compute directly (with or without TF-M enabled). This doesn't require secure storage support so we can simplify this even more. Please see related comments in this PR

Reply:
+1 to *not* have this if/case here. It is really bad if we start "ifdefing" because of TF-M.

Reply:
Following these comments I think that we can say that this PR is deprecated in favor of #71827 which only uses PSA functions